### PR TITLE
addpatch: ltrace, ver=0.7.3-5

### PR DESCRIPTION
**Pull request by YHStar**
- [Add loongarch64 support](https://github.com/YHStar/ltrace-0.7.3-5/commit/ad72dc9aa5e16f2e98f445aa79c847daea983df6.diff)
- [Fix some tests Headers for gcc14](https://github.com/YHStar/ltrace-0.7.3-5/commit/50791c5ac1e9bedaa4b67d55538dddf5b5afa710.diff)

### CODE REVIEW

File: update_config
Comment (wszqkzqk) [inline update_config:1]: Do not add packages with additional patches here. Please read [this](https://wszqkzqk.github.io/2024/08/12/loong-tools-design/#%E7%89%B9%E6%AE%8A%E6%83%85%E5%86%B5).

**Review by wszqkzqk — CHANGES_REQUESTED** [review]

File: ltrace/loong.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+diff --git a/PKGBUILD b/PKGBUILD
+index a568f2f..e99d606 100644
+--- a/PKGBUILD
++++ b/PKGBUILD
+@@ -15,6 +15,17 @@ backup=('etc/ltrace.conf')
+ source=(https://sources.archlinux.org/other/$pkgname/$pkgname-$pkgver.tar.bz2)
+ sha256sums=('0e6f8c077471b544c06def7192d983861ad2f8688dd5504beae62f0c5f5b9503')
+ 
++depends+=('autoreconf')
```
Comment (wszqkzqk) [inline ltrace/loong.patch:9]: Do not add this.

1. `autoreconf` is not a `pkgname` and cannot be installed. (It's `autoconf`)
2. You just run `autoreconf` to make the pkg. It's not a runtime dependency
3. Moreover, `autoconf` is in the `depends` of `base-devel`, so it's already in the build environment

**Review by wszqkzqk — CHANGES_REQUESTED** [review]

File: ltrace/loong.patch [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+index a568f2f..4bc7b4d 100644
+--- a/PKGBUILD
++++ b/PKGBUILD
+@@ -14,6 +14,15 @@ makedepends=('dejagnu')
+ backup=('etc/ltrace.conf')
+ source=(https://sources.archlinux.org/other/$pkgname/$pkgname-$pkgver.tar.bz2)
+ sha256sums=('0e6f8c077471b544c06def7192d983861ad2f8688dd5504beae62f0c5f5b9503')
++prepare(){
++  cd "$srcdir/$pkgname-$pkgver"
++  wget -O ./config/autoconf/config.sub "git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD"
```
Comment (wszqkzqk) [inline ltrace/loong.patch:11]: Would you please consider to add it to source?

Comment (YHStar) [reply ltrace/loong.patch]: Thanks for your sentence noticing me that my git isn't source, and I take all patches to the patches set.

**Comment by wszqkzqk** [comment]
Simplify. Please try this:

```diff
diff --git a/PKGBUILD b/PKGBUILD
index a568f2f..3aab670 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -37,4 +37,16 @@ package(){
   make DESTDIR="$pkgdir" install
 }
 
+prepare(){
+  cd "$srcdir/$pkgname-$pkgver"
+  patch -p1 -i ../loongarch64-support.patch
+  patch -p1 -i ../fix-tests.patch
+  autoreconf -fiv
+}
+
+source+=("loongarch64-support.patch::https://github.com/YHStar/ltrace-0.7.3-5/commit/ad72dc9aa5e16f2e98f445aa79c847daea983df6.diff"
+         "fix-tests.patch::https://github.com/YHStar/ltrace-0.7.3-5/commit/50791c5ac1e9bedaa4b67d55538dddf5b5afa710.diff")
+sha256sums+=('9ddd55ae96cb3aaa8ba1cd8f9dc7002569b77ee38af515e3df39ae180ef78320'
+             '1e12a754a5d173114b4bc2f3a2c3e79415d1d6d997d3c06effa4da0186df262a')
+
 # vim:set ts=2 sw=2 et:
```

**Review by wszqkzqk — APPROVED** [review]

**Pull request merged**